### PR TITLE
Add ai-env virtual package; remove python variant from jedi-base-env; bug fixes for Discover SCU16/17; doc updates for ParallelWorks; etc

### LINKED ISSUE #1012: Reminder: set env variable `I_MPI_EXTRA_FILESYSTEM='ON'` for all systems using intel@2021.8.0 and parallel filesystems
**Is your feature request related to a problem? Please describe.**
This environment variable existed in old versions of the Intel compiler, was removed, and was introduced again in intel@2021.8.0. This environment variable must be set if one wants to use parallel filesystems.

**Describe the solution you'd like**
See Hercules site config

**Additional context**
It is unclear what this environment variable does when jobs write to local filesystems (e.g. `/tmp`).

### PR DESCRIPTION
**Pull request by climbfuji**
### Summary

- Set env variable I_MPI_EXTRA_FILESYSTEM='ON' for Derecho and Discover SCU17
- Add diag-disable flag for deprecated Intel classic compilers for Hercules and Derecho
- Add protobuf and py-protobuf to list of module excludes
- Update README.md for noaa-aws, noaa-azure, noaa-gcloud
- Update documentation for Discover SCU16/SCU17
- Remove variant python from jedi-base-env
- Add ai-env virtual package and set py-torch variants in configs/common/packages.yaml
- Skip ai-env with Intel compilers in unified-dev template

### Testing

- [x] CI
- [x] @climbfuji's macOS
- [x] AWS ParallelCluster
- [x] Discover SCU16 (ai-env not tested)
- [x] Discover SCU17 (ai-env not tested)

### Applications affected

None

### Systems affected

- Discover
- Hercules, Derecho due to minor compiler config updates
- All systems when building `ai-env` with `gcc` or `clang`

### Dependencies

n/a

### Issue(s) addressed

Working towards https://github.com/JCSDA/spack-stack/issues/994
Working towards https://github.com/JCSDA/spack-stack/issues/1011 (two out of three ain't bad)
Resolves https://github.com/JCSDA/spack-stack/issues/1012

### Checklist
- [x] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so.
- [x] These changes have been tested on the affected systems and applications.
- [x] All dependency PRs/issues have been resolved and this PR can be merged.

### CODE REVIEW

File: configs/sites/derecho/compilers.yaml [hunk 1]
```diff
@@ -1,42 +1,47 @@
 compilers::
```
Comment (climbfuji) [inline configs/sites/derecho/compilers.yaml:1]: Except for lines 10-12 adding the annoying deprecation warnings, and line 27 adding the recently reintroduced I_MPI env var, the rest of the changes here are due to fixing the indentation only.

**Review by mathomp4 — APPROVED** [review]
Discover stuff looks good to me. I see new things I'll like to hear about! :)

File: configs/sites/noaa-azure/README.md [hunk 2]
```diff
@@ -6,7 +6,11 @@ that the site configuration files in `configs/sites/noaa-azure` work out of the
 
 sudo su
 chmod 777 /contrib
-yum install -y m4
```
Comment (natalie-perlin) [inline configs/sites/noaa-azure/README.md:9]: Does the new installation work successfully without m4?

Comment (natalie-perlin) [reply configs/sites/noaa-azure/README.md:9]: I think it was specific for Azure, not other platforms

Comment (climbfuji) [reply configs/sites/noaa-azure/README.md:9]: oops that's a mistake I think. Glad you are reviewing thoroughly!

Comment (climbfuji) [reply configs/sites/noaa-azure/README.md:9]: fixed

Comment (natalie-perlin) [reply configs/sites/noaa-azure/README.md:9]: Thank you!!

File: configs/sites/derecho/compilers.yaml (hunk 1 above)
Comment (natalie-perlin) [reply configs/sites/derecho/compilers.yaml:1]: These I_MPI_* variables might be needed for building or running the UFS and Apps as well. Probably this is something to keep in mind

**Review by RatkoVasic-NOAA — APPROVED** [review]
Tested on Hercules.
BTW, even with added comments for warnings in compilation log file is 1GB, 8M lines.
Approving.

**Review by natalie-perlin — APPROVED** [review]

File: configs/sites/derecho/compilers.yaml (hunk 1 above)
Comment (climbfuji) [reply configs/sites/derecho/compilers.yaml:1]: So these will be set automatically whenever the `stack-intel` compiler gets loaded on Derecho - should be sufficient for UFS

**Pull request merged**